### PR TITLE
Add scripts to generate csv files

### LINKED ISSUE #2: (i2b2) Clean original data files for ease of use
The source for this dataset has the training and test data split across many different files. We'll want to clean this all into two `train.csv` and `test.csv` files for ease of use.

### PR DESCRIPTION
**Pull request by bicycleman15**
closes #2

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+### Instructions to generate csv from i2b2 Obesity Challenge xml files
+
+First gain access to dataset. Once you have the access, you need to download the following xml files and rename exactly as below:
```
Comment (MarcioPorto) [inline scripts/README.md]: I am thinking that it might be better to read the files directly from the URL instead of requiring this manual step. We can include the access variables using environment variables to not expose the data. What do you think?

File: scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+2. Download Training Data: The second set of Obesity Training Records (XML) as rename as `text2.xml`
+3. Download Training Data: Intuitive Annotations for Training Records (XML) as rename as `label1.xml`
+4. Download Training Data: Annotations for the second set of Obesity Training Records (XML) as rename as `label2.xml`
+5. Download Training Data: Addendum to the Intuitive Annotations for Training Records (XML) as rename as `label3.xml`
+6. Download Test Data: Test Records (XML) as rename as `test_text.xml`
+7. Download Test Data: Ground Truth for Intuitive Judgments on Test Data (XML) as rename as `test_label.xml`
+
+Move all the files inside the `scripts` folder.
+
+Run `python gen_csv.py` to generate `train.csv` and `test.csv` and move them to the `data` folder.
```
Comment (MarcioPorto) [inline scripts/README.md]: Let's make it so the script already outputs the two files inside the data folder.

File: scripts/gen_csv.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
+import os
+from xml.etree import ElementTree
+import pandas as pd
```
Comment (MarcioPorto) [inline scripts/gen_csv.py:3]: Can you add a `requirements.txt` file with the project dependencies?

File: scripts/gen_csv.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+class Patient:
```
Comment (MarcioPorto) [inline scripts/gen_csv.py:5]: Really like that you added a class for this 🙂

File: scripts/gen_csv.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+dic = {} 
+
+# read the first training data
+tree = ElementTree.parse('./text1.xml')
+root = tree.getroot()
+docs  = root.findall('docs')[0]
+for doc in docs:
+    text_data = doc.findall('text')[0].text  # extract texxt data
+    pat = Patient() # create pateint
```
Comment (MarcioPorto) [inline scripts/gen_csv.py]: Can you fix the type in these two comments and the ones on lines 36 and 37?

File: scripts/gen_csv.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+# read the second training data
+tree = ElementTree.parse('./text2.xml')
+root = tree.getroot()
+docs  = root.findall('docs')[0]
+for doc in docs:
+    text_data = doc.findall('text')[0].text  # extract texxt data
+    pat = Patient() # create pateint
+    pat.text = text_data
+    pat.id = int(doc.attrib['id'])
+    dic[pat.id] = pat
```
Comment (MarcioPorto) [inline scripts/gen_csv.py]: This is essentially just a repetition of the above, so let's make this a loop instead

File: scripts/gen_csv.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+# read the annotations 3
+tree = ElementTree.parse('./label3.xml')
+root = tree.getroot()
+diseases  = root.findall('diseases')[0]
+for disease in diseases:
+    dis_name = disease.attrib['name'].split(' ')[0]
+    for doc in disease:
+        id_ = int(doc.attrib['id'])
+        setattr(dic[id_] , dis_name, doc.attrib['judgment'])
```
Comment (MarcioPorto) [inline scripts/gen_csv.py]: Same as above, let's make this a loop to avoid repeating the same steps in code.

File: scripts/gen_csv.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+    d_train['Hypercholesterolemia'].append(pat.Hypercholesterolemia)
+    d_train['Hypertension'].append(pat.Hypertension)
+    d_train['Hypertriglyceridemia'].append(pat.Hypertriglyceridemia)
+    d_train['OA'].append(pat.OA)
+    d_train['Obesity'].append(pat.Obesity)
+    d_train['OSA'].append(pat.OSA)
+    d_train['PVD'].append(pat.PVD)
+    d_train['Venous'].append(pat.Venous)
+
+data_train = pd.DataFrame(d_train)
```
Comment (MarcioPorto) [inline scripts/gen_csv.py]: If you add the above as a method on the `Patient` class, we can do this whole thing like this:

```
data_train = pd.DataFrame([pat.get_attrs() for pat in dic.values()])
```

`get_attrs()` could look something like:

```
def get_attrs(self):
    return {
        'id': self.id,
        'text': self. text,
        ...
    }
```

That way it's cleaner and we can also re-use it for the test set below.

File: scripts/gen_csv.py [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+data_train.to_csv('train.csv', sep=',') # spit the data to a csv
```
Comment (MarcioPorto) [inline scripts/gen_csv.py]: As I mentioned before, let's output this file into the `data` folder

File: scripts/gen_csv.py [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+
+dic_t = {}
```
Comment (MarcioPorto) [inline scripts/gen_csv.py]: For all the test data processing below, you can use the same process/helper methods I outlined above 🚀

**Review by MarcioPorto — CHANGES_REQUESTED** [review]
Thanks for getting this started @bicycleman15! Left a few refactoring notes for you.

File: scripts/README.md (hunk 1 above)
Comment (bicycleman15) [reply scripts/README.md]: Yes, sounds good. I am having a little problems implementing this. Can you help me here ?

Comment (MarcioPorto) [reply scripts/README.md]: Yeah, absolutely. What is the issue?

File: scripts/gen_csv.py [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    text_data = doc.findall('text')[0].text
+    
+    # create pateint
+    pat = Patient()
+
+    pat.text = text_data
+    pat.id = int(doc.attrib['id'])
+
+    # store in dic
+    dic_t[pat.id] = pat
```
Comment (MarcioPorto) [inline scripts/gen_csv.py]: These lines are still the same as what you used above to process the training data. Let's move this to a helper method that takes in a file path and does the processing so we can re-use it here.

Same idea for the testing logic below.

**Review by MarcioPorto — CHANGES_REQUESTED** [review]
One more small change and we're off 🚀

File: scripts/gen_csv.py [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+                'obesity_standoff_annotations_training_addendum.xml']
+
+for file in train_labels:
+    # read the annotations
+    tree = ElementTree.parse('../downloads/' + file)
+    root = tree.getroot()
+    diseases  = root.findall('diseases')[0]
+    for disease in diseases:
+        dis_name = disease.attrib['name'].split(' ')[0]
+        print(dis_name)
```
Comment (MarcioPorto) [inline scripts/gen_csv.py]: Remove this print statement if it's not needed anymore. If you do want to log it, let's use a logger instead.

**Review by MarcioPorto — APPROVED** [review]
LGTM @bicycleman15 🚀
